backend/src/contaxy/managers/deployment/manager.py

```diff
@@ -1,16 +1,261 @@
-from contaxy.schema import JobInput, ResourceAction, ServiceInput
+from contaxy.operations import DeploymentOperations, JsonDocumentOperations
+from contaxy.schema import (
+    Job,
+    JobInput,
+    ResourceAction,
+    ResourceNotFoundError,
+    Service,
+    ServiceInput,
+)
+from contaxy.schema.deployment import DeploymentStatus, DeploymentType
 
 ACTION_DELIMITER = "-"
 ACTION_ACCESS = "access"
 
 
-class DeploymentManager(DeploymentOperations):
+def _get_service_collection_id(project_id: str) -> str:
+    return f"project_{project_id}_service_metadata"
+
+
+def _get_job_collection_id(project_id: str) -> str:
+    return f"project_{project_id}_job_metadata"
+
+
+class DeploymentManagerWithDB(DeploymentOperations):
+    def __init__(
+        self,
+        deployment_manager: DeploymentOperations,
+        json_db_manager: JsonDocumentOperations,
+    ):
+        self.deployment_manager = deployment_manager
+        self.json_db = json_db_manager
+
+    def deploy_service(
+        self,
+        project_id: str,
+        service: ServiceInput,
+        action_id: Optional[str] = None,
+        deployment_type: Literal[
+            DeploymentType.SERVICE, DeploymentType.EXTENSION
+        ] = DeploymentType.SERVICE,
+    ) -> Service:
+        deployed_service = self.deployment_manager.deploy_service(
+            project_id, service, action_id, deployment_type
+        )
+        self.json_db.create_json_document(
+            project_id=config.SYSTEM_INTERNAL_PROJECT,
+            collection_id=_get_service_collection_id(project_id),
+            key=deployed_service.id,
+            json_document=deployed_service.json(),
+            upsert=False,
+        )
+        return deployed_service
+
+    def list_services(
+        self,
+        project_id: str,
+        deployment_type: Literal[
+            DeploymentType.SERVICE, DeploymentType.EXTENSION
+        ] = DeploymentType.SERVICE,
+    ) -> List[Service]:
+        service_docs = self.json_db.list_json_documents(
+            project_id=config.SYSTEM_INTERNAL_PROJECT,
+            collection_id=_get_service_collection_id(project_id),
+        )
+        deployed_service_lookup: Dict[str, Service] = {
+            service.id: service
+            for service in self.deployment_manager.list_services(
+                project_id, deployment_type
+            )
+            if service.id is not None
+        }
+        services = []
+        # Go through all services in the DB and update their status
+        for service_doc in service_docs:
+            db_service = Service.parse_raw(service_doc.json_value)
+            deployed_service = deployed_service_lookup.pop(db_service.id, None)
+            if deployed_service is None:
+                db_service.status = DeploymentStatus.STOPPED
+            else:
+                db_service.status = deployed_service.status
+            services.append(db_service)
+        # Add services to DB that were not added via the contaxy API
```

Comment:
I am wondering whether this is the best approach here or whether it should not be possible to add services not via the Contaxy API **or** if it is possible, whether to add an extra endpoint to add it to the db. 
this way, no implicit lookup logic would need to be made here and the DB contains all the truth. what do you think?

Reply:
I agree, that it would definitely be more consistent if we don't allow this. At the moment I needed this feature, because I am using docker compose (and helm) to start all ML Lab components. This includes the workspace extension which needs to be picked up by contaxy. For now, I really like that I can start everything with docker compose.
If at a later point we have a proper UI for installing extensions, we could maybe remove this feature? What would you say?

Reply:
alrighty, that makes sense